registration/src/main/java/com/kyriba/registration/api/AttendeeController.java

```diff
@@ -0,0 +1,78 @@
+package com.kyriba.registration.api;
+
+import com.kyriba.registration.api.dto.AttendeeTransition;
+import com.kyriba.registration.api.dto.ExchangeTransition;
+import com.kyriba.registration.domain.TicketStatus;
+import lombok.Value;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_UTF8_VALUE;
+
+
+@RestController
+@RequestMapping("/api/v1/attendees")
+public class AttendeeController
+{
+  /**
+   * UCS: Attendee: 1. register in a conference
+   *
+   * @param attendee input API DTO for new attendee registration
+   * @return registered attendee output API DTO
+   */
+  @PostMapping(produces = APPLICATION_JSON_UTF8_VALUE, consumes = APPLICATION_JSON_UTF8_VALUE)
+  AttendeeResponse register(@RequestBody AttendeeTransition attendee)
+  {
+    return new AttendeeResponse("123");
+  }
+
+
+  /**
+   * UCS: Attendee: 5. see (ticket) registration status
+   *
+   * @param id attendees id
+   * @return registration status output API DTO
+   */
+  @GetMapping(value = "/{id}/status", produces = APPLICATION_JSON_UTF8_VALUE)
+  StatusResponse status(@PathVariable String id)
+  {
+    return new StatusResponse(TicketStatus.NOT_PAID);
+  }
+
+
+  /**
+   * UCS: Attendee: 4. change attendee
+   *
+   * @param exchange input API DTO for exchanging ticket
+   * @return registered attendee output API DTO
+   */
+  @PostMapping(value = "/{id}/exchange", produces = APPLICATION_JSON_UTF8_VALUE, consumes = APPLICATION_JSON_UTF8_VALUE)
+  AttendeeResponse exchange(@PathVariable String id, @RequestBody ExchangeTransition exchange)
```

Comment:
Probably this logic should be in another microserviceservice, because both attendees have to be registered and this operation more related to their statuses and ticket ownership.

Reply:
my idea is:
POST /api/v1/attendees/123/exchange
JSON 
```
{
  "newTicketOwner": {
    "firstName": "Ivan",
    "lastName": "Sidorov",
    "email": "ivan.sid@gmail.com",
    "mobilePhone": "+375991234567",
    "job": {
      "company": "Kyriba Corp.",
      "position": "Senior Software Engineer",
      "city": "Minsk"
    }
  }
}
```
Attendee with id=123 is registered, and in posted JSON we provide "newTicketOwner" - a new attendee to be registered as an owner of the ticket.

Do you mean that is's better to have both attendees already registered and delegate this logic to an other microservice (e.g. Ticket service)?

Reply:
Exactly!